usersession/userd/privileged_desktop_launcher.go

```diff
@@ -118,6 +129,20 @@ func (s *PrivilegedDesktopLauncher) OpenDesktopEntry(desktopFileID string, sende
 	return nil
 }
 
+// argumentsSecurityCheck ensures that none of the arguments passed
+// by begins with "-", to avoid passing parameters to the application.
```

Comment:
I think we want to be more opinionated here, and expect the arguments to be URLs. I realise that this will be difficult for the client side to handle before we change how the the client knows whether it is being called with URLs or file names, but it will put the server side in a better position to interpret the arguments.

So we'd probably want to check:
 * `url.Parse` doesn't return an error
 * parsed URL scheme is not empty
 * If scheme is `file`, that path starts with `/`

This should ensure that we don't generate arguments beginning with a dash.

Reply:
Done.